### ISSUE TITLE
[BB]Mag evolution desync with client.

### ISSUE THREAD

**Issue by collett8192** [fuzziqersoftware/newserv#497]
**To reproduce**
1. Create a RA character.
2. Feed mag with Antidote until level 35 (and see the mag evolve on client side).
3. Re-login.
4. Mag evolution reverted.

**Game version(s) (choose one or more of the following):**
BB

**Server log output**
N/A

**Additional context**
Upon inspecting [the code](https://github.com/fuzziqersoftware/newserv/blob/master/src/Items.cc#L354), in order to evolve into mag 0x02 the flag 0x10 (DEX highest) cannot be set, which is the opposite of what is expected (DEX should be the highest). By looking at code below I believe the ` == 0` is a mistake, it should be `if (flags & 0x110)`.

But there're more problems. The `mag_item.compute_mag_strength_flags()` function actually cannot produce any value > 0x100 (only 4 possible: 0x08, 0x10, 0x20, 0x100), so `flags & 0x110` will never be true. In order to make it work I had to modify it to just `flags & 0x10`, which I believe is the correct condition for this particular evolution.

There are more code checking the flag with 0x100 + something down the line and will not work due to this bug.

**Comment by fuzziqersoftware** [comment]
Thanks for investigating this. The `== 0` is indeed a mistake; I had switched the order of the cases to make the code more readable after reverse-engineering this, and forgot to invert the condition. This is now fixed, but I haven't had a chance to test it myself yet.

The actual flags used should be correct; it's true that `compute_mag_strength_flags` should never return a value > 0x100, but the condition `flags & 0x110` checks for either `flags == 0x100` or `flags == 0x010`.
